core/arch/arm/kernel/tee_ta_manager.c

```diff
@@ -109,6 +109,33 @@ static void set_tee_rs(struct tee_ta_session *tee_rs)
 	thread_set_tsd(tee_rs);
 }
 
+static void tee_ta_set_busy(struct tee_ta_ctx *ctx)
+{
+	while (true) {
+		bool old_busy;
+
+		mutex_lock(&tee_ta_mutex);
+		old_busy = ctx->busy;
+		if (!old_busy)
```

Comment:
Sure, we can remove the if. `old_busy` is needed since I'd like to match the `mutex_lock()` with one `mutex_unlock()` and I need to know if some other thread already holds `ctx->busy`.